### ISSUE TITLE
refine forward

### ISSUE THREAD

**Issue by kongbia** [foolwood/SiamMask#174]
```
      if test:
            p0 = torch.nn.functional.pad(f[0], [16, 16, 16, 16])[:, :, 4*pos[0]:4*pos[0]+61, 4*pos[1]:4*pos[1]+61]
            p1 = torch.nn.functional.pad(f[1], [8, 8, 8, 8])[:, :, 2 * pos[0]:2 * pos[0] + 31, 2 * pos[1]:2 * pos[1] + 31]
            p2 = torch.nn.functional.pad(f[2], [4, 4, 4, 4])[:, :, pos[0]:pos[0] + 15, pos[1]:pos[1] + 15]
        else:
            p0 = F.unfold(f[0], (61, 61), padding=0, stride=4).permute(0, 2, 1).contiguous().view(-1, 64, 61, 61)
            if not (pos is None): p0 = torch.index_select(p0, 0, pos)
            p1 = F.unfold(f[1], (31, 31), padding=0, stride=2).permute(0, 2, 1).contiguous().view(-1, 256, 31, 31)
            if not (pos is None): p1 = torch.index_select(p1, 0, pos)
            p2 = F.unfold(f[2], (15, 15), padding=0, stride=1).permute(0, 2, 1).contiguous().view(-1, 512, 15, 15)
            if not (pos is None): p2 = torch.index_select(p2, 0, pos)
```
Can this line ```if not (pos is None): p0 = torch.index_select(p0, 0, pos)``` be deleted because there is no ```pos``` in the training?

**Comment by zosel260** [comment]
@kongbia 
This codes must be remained because pos means positive samples and pos exists during training also.

**Comment by kongbia** [comment]
> @kongbia
> This codes must be remained because pos means positive samples and pos exists during training also.

but in siammask_sharp.py line 65, there is no pos input during training.
Pos in training comes from the mask_weight in siam_mask_dataset.py, and is used only in loss function.
